src/app/core/export/export-service/export.service.ts

```diff
@@ -34,34 +35,100 @@ export class ExportService {
    * @param config (Optional) config specifying which fields should be exported
    * @returns string a valid CSV string of the input data
    */
-  createCsv(data: any[], config?: ExportColumnConfig[]): string {
-    const allFields = new Set();
+  async createCsv(data: any[], config?: ExportColumnConfig[]): Promise<string> {
     const exportableData = [];
+    const columns = new Set<string>(
+      (config ?? []).map((c) => c.label ?? c.query)
+    );
 
-    data.forEach((element: any) => {
-      const exportableObj = {};
-
-      const currentRowConfig =
+    for (const dataRow of data) {
+      const rowConfig =
         config ??
-        Object.keys(element).map((key) => ({ key: key } as ExportColumnConfig));
-      for (const columnConfig of currentRowConfig) {
-        const label = columnConfig.label ?? columnConfig.key;
-        const value = entityListSortingAccessor(element, columnConfig.key);
-        if (value?.toString().match(/\[object.*\]/) !== null) {
-          // skip object values that cannot be converted to a meaningful string
-          continue;
-        }
-
-        exportableObj[label] = value;
-        allFields.add(label);
+        Object.keys(dataRow).map(
+          (key) => ({ query: key } as ExportColumnConfig)
+        );
```

Comment:
I feel like it would make things much easier if we also refer to the property schemas of the entities here. The labels of the properties can be reused and we dont have to worry about entities that are missing some properties, so the `rowConfig` and `columns` dont have to be updated for every `dataRow`. 
For the export buttons on the table, it will always be just one entity and for the general export we would just have to iterate over every entity and create the export for the entity.